examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -59,8 +58,12 @@
     "ScreenPorch",
 ]
 
-X = X[categorical_columns_subset + numerical_columns_subset]
-X[categorical_columns_subset] = X[categorical_columns_subset].astype("category")
+# Comment the line below to run the example on the full dataset:
+X = X[column_subset]
+
+# Explicitly type the categorical columns as such.
+object_dtyped_columns = X.select_dtypes(include=["object"]).columns
+X[object_dtyped_columns] = X[object_dtyped_columns].astype("category")
```

Comment:
I changed this because of: https://github.com/scikit-learn/scikit-learn/pull/24907/files#r1020949896